File: 149. Max Points on a Line/solution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int gcd(int first, int second){
        if(second ==0) return first;
        return gcd(second, first%second);
    }
    
    int maxPoints(vector<vector<int>> &points){
        int maxPoints = 0;


        for(int i=0; i<points.size(); i++){
            map<pair<int, int>, int> HM;
            int duplicate = 0;
            for(int j=i+1; j<points.size(); j++){
                int dy = points[i][1] - points[j][1];
                int dx = points[i][0] - points[j][0];

                if(dy==0 && dx==0)
                    duplicate += 1;
                else{
                    int GCD = __gcd(abs(dy), abs(dx));
                    dy /= GCD;
                    dx /= GCD;
                    HM[{dy, dx}] += 1;
                }

            }
            for(auto it: HM){
                maxPoints = max(maxPoints, it.second);
            }
        }
        return maxPoints;
    }
};

int main(){
    
}

File: 149. Max Points on a Line/solution.java
import java.util.HashMap;
import java.util.Set;

import jdk.internal.net.http.common.Pair;

// Given an array of points where points[i] = [xi, yi] represents a point on the X-Y plane, return the maximum number of points that lie on the same straight line.

class Solution {
    public int maxPoints(int[][] points) {
        HashMap<Pair<Integer, Integer>, Set<Pair<Integer, Integer>>> HM = new <Pair<Integer, Integer>, Set<Pair<Integer, Integer>>>();
        
        for(int i=0; i<points.length; i++){
            int x = points[i][0];
            int y = points[i][1];

            
        }
    }
}